### PR TITLE
Update mirador config from mirador

### LINKED ISSUE #88: Add "Preview and configure viewer" button/popup to mirador widget
Larger design ticket: sul-dlss/exhibits#1002

The popup should persist the Mirador state in the field provided by #69 

widget-ui

popup-ui

### LINKED ISSUE #93: Figure out how to pass over mirador state to embeded mirador viewer
Discussed in https://github.com/sul-dlss/vatican_exhibits/pull/89

### LINKED ISSUE #186: Get config from only the selected Mirador Sir Trevor block
Raised by @aeschylus here: https://github.com/sul-dlss/vatican_exhibits/pull/160#discussion_r199235812

This needs to be scoped to only the requested SirTrevor block.

### PR DESCRIPTION
**Pull request by mejackreed**
Fixes #88
Fixes #93 
Fixes #186 
Some rough issues here where follow on tickets are needed but basic concepts all working.

Can view in action here: https://drive.google.com/file/d/1WjeeUukMKnze1i9Tj4EWVNcLV_IOGdxx/view

### CODE REVIEW

File: app/assets/javascripts/mirador_widget_admin.js [hunk 1]
```diff
@@ -43,6 +49,35 @@
...
+              canvasID: value.canvasID,
+              loadedManifest: value.loadedManifest,
+              sidePanelVisible: value.sidePanelVisible,
+              windowOptions: value.windowOptions ? { osdBounds: value.windowOptions.osdBounds } : undefined
+            }
+          })
+        };
+        // Add config to hidden form.
+        var input = '<input type="text" style="display:none;" name="mirador_config" value=\'' +
+         JSON.stringify(newConfig) +
```
Comment (cbeer) [inline app/assets/javascripts/mirador_widget_admin.js]: I'm immediately suspicious of this and wonder if there are any data escaping concerns we should be concerned about?

Comment (mejackreed) [reply app/assets/javascripts/mirador_widget_admin.js]: Should be fixed now in bb681c9864227114dd10456d55833cbc4ba035f5 . Thanks for the suggestion!

File: app/assets/javascripts/mirador_messager.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var miradorInstance;
+$(document).on('ready turbolinks:load', () => {
+  $('[data-miradorfromscratch]').each(function(i, value) {
+    
+    // Setup Mirador to use config already existing in parent form
+    parent.$('#mirador-modal').on('shown.bs.modal', function () {
+      var miradorConfig = JSON.parse(parent.$('[name="mirador_config"]').val());
```
Comment (aeschylus) [inline app/assets/javascripts/mirador_messager.js]: This actually gives the same configuration to all viewers on the page.

Comment (mejackreed) [reply app/assets/javascripts/mirador_messager.js]: @aeschylus mind creating a spin off ticket so we can come back to this?

**Pull request merged**